### ISSUE TITLE
SignalsExp as subclass

### ISSUE THREAD

**Issue by k1o0** [cortex-lab/Rigbox#212]
MATLAB's moved on considerably since Chris implemented his performance tweaks to SignalsExp for performance.  Once a performance test is in place for SignalsExp (#66) we can test the impact to performance when returning to a more intuitive OOP design, namely restoring use of hw.Window in SignalsExp and reducing code redundancy by making SignalsExp a subclass of Experiment.  I expect the former may have the bigger impact on performance.

**Comment by jaib1** [comment]
what were Chris' tweaks to `exp.SignalsExp` and how did they improve performance? I gather one was having `srv.expServer` create the PTB Screen rather than `exp.SignalsExp`?

**Comment by k1o0** [comment]
>  I gather one was having `srv.expServer` create the PTB Screen rather than `exp.SignalsExp`

No, expServer always initialized the hardware, including the PTB Window.

> what were Chris' tweaks to `exp.SignalsExp` 

There were many but mostly it was reducing the number of method calls because this used to be [very slow](https://stackoverflow.com/questions/1693429/is-matlab-oop-slow-or-am-i-doing-something-wrong#1745686).
